ros2cli/completion/ros2-argcomplete.fish

```diff
@@ -0,0 +1,19 @@
+# Copyright 2017-2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if type register-python-argcomplete3 > /dev/null 2>&1
+  eval "register-python-argcomplete3 --shell fish ros2 | source"
```

Comment:
Using `eval` and `source` looks weird - are both really necessary?

Reply:
The `.` command is deprecated as the `source` command is preferred: fish-shell/fish-shell#310
But I suppose the eval might not be needed. I was just mimicking the style in bash file.
Why was eval used there?

Reply:
Because it is the recommended approach described in the `argcomplete` docs: https://argcomplete.readthedocs.io/en/latest/#synopsis